### ISSUE TITLE
Issue while parsing Config.groovy

### ISSUE THREAD

**Issue by gianlucagenga** [grails/grails-data-mapping#782]
Hi,
I have an issue since I updated plugin ":mongodb:3.0.3" to 6.0.0RC1 on Grails 2.5.4, in order to use MultiDatasource and MultiTenancy.
As I can see, the Config.groovy file is not parsed correctly. Let me give you an example:
Given:
propA {
    value1 = 'value1'
    value2 = 'value2
}

the result is a LinkedHashMap containing 
"propA": { value1: "value1", value2: "value2" }
"propA.value1": 'value1'
"propB.value2": 'value2'

while I was just expecting "propA": { value1: "value1", value2: "value2" }.
Version 6.0.0M1 resolves the issue, but has no support for MultiDatasource since unuseful for my purposes.

Can you help me please?
Thanks

**Comment by graemerocher** [comment]
How does this manifest as an actual problem? Please upload an example that reproduces the issue

**Comment by gianlucagenga** [comment]
I'll try to be more specific. The problem is very easy to reproduce.
I have these lines in my Config.groovy
`oauth {
   providers {
       google {
           api = it.h14.providers.GoogleApi20
           scope = 'openid email'
           callback   = "${grails.serverURL}/oauthCallBack/callback"
           successUri = "${grails.serverURL}/oauthCallBack/googleSuccess"
           failureUri = "${grails.serverURL}/oauthCallBack/googleFail"
       }
   }
}`

then I debug my application and watch inside the grailsApplication injected bean.
`grailsApplication.config.oauth.providers` contains 6 keys
1) "google" = [ "api": it.h14.providers.GoogleApi20, "scope": 'openid email' .... ]
2) "google.api" =  it.h14.providers.GoogleApi20
3) "google.callback" = 'openid email'
4) "google.scope" = "http://localhost:8080/oauthCallBack/callback"
5) "google.successUri" = "http://localhost:8080/oauthCallBack/googleSuccess"
6) "google.failureUri"=  "http://localhost:8080/oauthCallBack/googleFail"

Basically, as you can see there are duplicated keys (2nd to 6th), while I was just expecting the first one containig all of them. This was the common behaviour before upgrading to 6.0.0RC1.
This causes me problems as the application loops on `grailsApplication.config.oauth.providers`

Hope everything is clear now.
Thank you

**Comment by graemerocher** [comment]
I don't see how this relates to GORM, are you saying GORM is mutating your configuration?

**Comment by gianlucagenga** [comment]
I agree with you, this sounds strange to me too. But I haven't changed anything else in my src code expect for the gorm update in BuildConfig.groovy.
It seems GORM changes the way json configuration is parsed. 
The problem is only on 6.0.0RC1, while 6.0.0M1 works perfectly.

Thank you

**Comment by Travis9011** [comment]
I have the same problem! 

I created a test project. You only need to uncomment `compile ":mongodb:6.0.0.RC1"` in a BuildConfig and comment the M1 version.

https://github.com/Travis9011/TestGORM6

**Comment by graemerocher** [comment]
Once the build https://travis-ci.org/grails/grails-data-mapping/builds/159242679 completes, please add the following dependency to your `BuildConfig.groovy` and verify if the fix works for you.

```
     compile "org.grails:grails-datastore-core:6.0.0.BUILD-SNAPSHOT", {
            transitive = false
        }
```
